lib/cosmos/dart/lib/dart_common.rb

```diff
@@ -0,0 +1,527 @@
+      if item.data_type == :DERIVED
+        next unless well_defined_read_conversion(item)
+      end
+
+      raw_data_type, converted_data_type = get_db_types(item)
+      item_id = lookup_item_id(packet_id, item.name)
+      if separate_raw_con?(item)
+        # Need both RAW and CONVERTED
+        ItemToDecomTableMapping.create(
+          :item_id => item_id,
+          :value_type => ItemToDecomTableMapping::RAW,
+          :reduced => REDUCED_TYPES.include?(raw_data_type),
+          :packet_config_id => packet_config.id,
+          :table_index => item_index / MAX_COLUMNS_PER_TABLE,
+          :item_index => item_index
+        )
+        item_index += 1
+        data_types << raw_data_type
+        ItemToDecomTableMapping.create(
+          :item_id => item_id,
+          :value_type => ItemToDecomTableMapping::CONVERTED,
+          :reduced => REDUCED_TYPES.include?(converted_data_type),
+          :packet_config_id => packet_config.id,
+          :table_index => item_index / MAX_COLUMNS_PER_TABLE,
+          :item_index => item_index
+        )
+        item_index += 1
+        data_types << converted_data_type
+      else
+        # Can combine RAW and CONVERTED (RAW_CON)
+        ItemToDecomTableMapping.create(
+          :item_id => item_id,
+          :value_type => ItemToDecomTableMapping::RAW_CON,
+          :reduced => REDUCED_TYPES.include?(raw_data_type),
+          :packet_config_id => packet_config.id,
+          :table_index => item_index / MAX_COLUMNS_PER_TABLE,
+          :item_index => item_index
+        )
+        item_index += 1
+        data_types << raw_data_type
+      end
+    end
+
+    return data_types
+  end
+
+  # Create the packet configuration in the database. This includes the
+  # ItemToDecomTableMapping which maps a telemetry item to its location in the
+  # decommutation table as well as the actual decommutation tables which hold
+  # the decommutated data. These tables are all named tXXX_YYY where XXX is the
+  # PacketConfig ID and YYY is a table index when the packet has more than
+  # MAX_COLUMNS_PER_TABLE and must span multiple tables. In addition the
+  # reduction tables are setup with the '_m' (minute), '_h' (hour), '_d' (day)
+  # extensions added to the tXXX_YYY tables (assuming the packets contain
+  # data which can be reduced.)
+  #
+  # @param packet [Cosmos::Packet] Packet to create database tables for
+  # @param packet_id [Integer] Id in the Packet table
+  # @param packet_config [PacketConfig] ActiveRecord access to the PacketConfig table
+  def setup_packet_config(packet, packet_id, packet_config)
+    data_types = setup_item_to_decom_table_mapping(packet, packet_id, packet_config)
+
+    # Create the actual decom table(s)
+    table_index = 0
+    data_types.each_slice(MAX_COLUMNS_PER_TABLE) do |table_data_types|
+      begin
+        ActiveRecord::Base.connection.drop_table("t#{packet_config.id}_#{table_index}")
```

Comment:
It looks like setup_packet_config is called by cleanup which happens every time DART starts. Isn't this code dropping the decommutation tables? How does this not clear the DB every time DART starts?

Reply:
It is only called for PacketConfigs that are not ready.